Original patch from Gentoo:
gentoo/src/patchsets/binutils/2.19/78_all_use-relro.patch

diff -durN binutils-2.19.orig/ld/ldmain.c binutils-2.19/ld/ldmain.c
--- binutils-2.19.orig/ld/ldmain.c	2008-11-23 16:59:24.000000000 +0100
+++ binutils-2.19/ld/ldmain.c	2008-11-23 16:59:45.000000000 +0100
@@ -264,6 +264,7 @@
   link_info.keep_memory = TRUE;
   link_info.new_dtags = TRUE;
   link_info.combreloc = TRUE;
+  link_info.relro = TRUE;
   link_info.strip_discarded = TRUE;
   link_info.emit_hash = TRUE;
 #ifndef __mips__
